modules/photons_control/clean.py

```diff
@@ -0,0 +1,212 @@
+
+    .. code-block:: python
+
+        await target.send(ChangeCleanCycle(enable=True), ["d073d5000001", "d073d5000001"])
+
+    Options are:
+
+        enable - boolean (required)
+            Pass True to start a cleaning cycle or False to stop a cycle in progress
+
+        duration - integer (optional)
+            Only used if enable=True. Specifies the duration (in seconds) for
+            the cleaning cycle. If not specified, the default duration will be
+            used. Default duration can be set using target:set_clean_config and
+            returned using target_get_clean_config
+
+    """
+
+    async def gen(reference, sender, **kwargs):
+        plans = sender.make_plans("capability")
+        async for serial, _, info in sender.gatherer.gather(plans, reference, **kwargs):
+            if info["cap"].has_hev:
+                yield LightMessages.SetHevCycle(
+                    enable=enable,
+                    res_required=False,
+                    ack_required=True,
+                    duration_s=duration,
+                    target=serial,
+                )
+
+    return FromGenerator(gen)
+
+
+@task
+class start_clean_cycle(task.Task):
+    """
+    Start a cleaning cycle on an HEV-capable bulb
+
+    ``target:start_clean_cycle match:cap=hev -- '{"duration": 7200}'``
+
+    It  takes a ``duration`` field that is the seconds of the duration. If not
+    provided, or set to 0, the default duration will be used.
+    """
+
+    target = task.requires_target()
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        options = sb.dictionary_spec().normalise(Meta.empty(), self.photons_app.extra_as_json)
+        await self.target.send(ChangeCleanCycle(enable=True, **options), self.reference)
+
+
+@task
+class stop_clean_cycle(task.Task):
+    """
+    Stop a cleaning cycle on an HEV-capable bulb
+
+    ``target:stop_clean_cycle match:cap=hev``
+
+    """
+
+    target = task.requires_target()
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        options = sb.dictionary_spec().normalise(Meta.empty(), self.photons_app.extra_as_json)
+        await self.target.send(ChangeCleanCycle(enable=False, **options), self.reference)
+
+
+@task
+class get_clean_status(task.Task):
+    """
+    Get the current cleaning status from the specified bulb.
+    """
+
+    target = task.requires_target()
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        async with self.target.session() as sender:
+            plans = sender.make_plans("hev_status")
+            async for serial, _, info in sender.gatherer.gather(plans, self.reference, **kwargs):
+                if info is not Skip:
+                    print(serial)
+                    if info["active"]:
+                        power_off = "yes" if info["power_off"] else "no"
+                        if humanize:
```

Comment:
if you make a function that returns this then you don't have to repeat the usage of humanize api